classes/submission/maps/Schema.php

```diff
@@ -187,8 +193,9 @@ public function mapMany(Enumerable $collection, LazyCollection $userGroups, arra
      *
      * @param LazyCollection<int,UserGroup> $userGroups The user groups in this context
      * @param Genre[] $genres The file genres in this context
+     * @param bool|Collection<int> $anonymizeReviews List of review assignment IDs to anonymize
      */
-    public function summarizeMany(Enumerable $collection, LazyCollection $userGroups, array $genres): Enumerable
+    public function summarizeMany(Enumerable $collection, LazyCollection $userGroups, array $genres, bool|Collection $anonymizeReviews = false): Enumerable
```

Comment:
Yes, we already have all or almost all data. From stage assignments we derive whether all recommendations are in and editors assigned

Reply:
And its not enough to decide also on anonymisation - that was my question. Instead of having extra fetches and logic in anonymizeReviews()?

Reply:
Alright..  I noticed there is already some anonymise logic within Schema class

```
        if (in_array('publications', $props)) {
            $currentUserReviewAssignment = Repo::reviewAssignment()->getCollector()
                ->filterBySubmissionIds([$submission->getId()])
                ->filterByReviewerIds([$this->request->getUser()->getId()])
                ->filterByLastReviewRound(true)
                ->getMany()
                ->first();
            $anonymize = $currentUserReviewAssignment && $currentUserReviewAssignment->getReviewMethod() === ReviewAssignment::SUBMISSION_REVIEW_METHOD_DOUBLEANONYMOUS;
        }

```
And it still could be controlled from outside - by enabling/disabling reviewAssignment anonymisation - and it could be probably quite simple check within the schema, whether reviewAssignment is eligible to be applied.
I believe it would be just checking whether there is stage assignment with matching current user id and role author?

Reply:
Ideally, I think such tasks should be handled by specific policies. Eventually we are planning to move towards Laravel Models and can use something like a policy method to give access to submission properties to specific roles https://laravel.com/docs/11.x/authorization#policy-methods

Reply:
> Alright.. I noticed there is already some anonymise logic within Schema class

The anonymization logic for publication data is done not in the schema but in the `PKPSubmissionController`.